controllers/activemqartemis_reconciler.go

```diff
@@ -1096,95 +1199,54 @@ func formatTemplatedString(customResource *brokerv1beta1.ActiveMQArtemis, templa
 	return template
 }
 
-func generateConsoleSSLFlags(customResource *brokerv1beta1.ActiveMQArtemis, namer common.Namers, client rtclient.Client, secretName string) string {
+func (r *ActiveMQArtemisReconcilerImpl) generateCommonSSLFlags(customResource *brokerv1beta1.ActiveMQArtemis, labels map[string]string, secretName string, caSecretName *string, keyStoreType string, trustStoreType string, client rtclient.Client, currentSS *appsv1.StatefulSet, isConsole bool) (*certutil.SslArguments, string, error) {
 
-	sslFlags := ""
 	secretNamespacedName := types.NamespacedName{
 		Name:      secretName,
 		Namespace: customResource.Namespace,
 	}
-	namespacedName := types.NamespacedName{
-		Name:      customResource.Name,
-		Namespace: customResource.Namespace,
-	}
+
 	stringDataMap := map[string]string{}
-	userPasswordSecret := secrets.NewSecret(namespacedName, secretName, stringDataMap, namer.LabelBuilder.Labels())
+	sslSecret := secrets.NewSecret(secretNamespacedName, stringDataMap, labels)
 
-	keyStorePassword := "password"
-	keyStorePath := "/etc/" + secretName + "-volume/broker.ks"
-	trustStorePassword := "password"
-	trustStorePath := "/etc/" + secretName + "-volume/client.ts"
-	if err := resources.Retrieve(secretNamespacedName, client, userPasswordSecret); err == nil {
-		if string(userPasswordSecret.Data["keyStorePassword"]) != "" {
-			keyStorePassword = string(userPasswordSecret.Data["keyStorePassword"])
-		}
-		if string(userPasswordSecret.Data["keyStorePath"]) != "" {
-			keyStorePath = string(userPasswordSecret.Data["keyStorePath"])
-		}
-		if string(userPasswordSecret.Data["trustStorePassword"]) != "" {
-			trustStorePassword = string(userPasswordSecret.Data["trustStorePassword"])
+	if err := resources.Retrieve(secretNamespacedName, client, sslSecret); err != nil {
+		return nil, "", err
+	}
+
+	// for trust manager ca bundle.
+	// in user secret case it can have both keystore and truststore so a separate ca bundle is optional
+	var caSecret *corev1.Secret = nil
+	if caSecretName != nil {
+		trustSecretNamespacedName := types.NamespacedName{
+			Name:      *caSecretName,
+			Namespace: customResource.Namespace,
 		}
-		if string(userPasswordSecret.Data["trustStorePath"]) != "" {
-			trustStorePath = string(userPasswordSecret.Data["trustStorePath"])
+
+		caSecret = secrets.NewSecret(trustSecretNamespacedName, stringDataMap, labels)
```

Comment:
This initialization could be avoid because the function Retrieve also works with empty resources, see my previous comment.

Reply:
+1